api/src/Entity/Discussion.php

```diff
@@ -25,23 +27,32 @@
 
 #[ORM\Entity(repositoryClass: DiscussionRepository::class)]
 #[ApiResource(
+    operations: [
+        new GetCollection(security: "is_granted('IS_AUTHENTICATED_FULLY')"),
+        new Get(security: "is_granted('ROLE_ADMIN') or object.customer == user or user.isAssociatedWithRepairer(object.repairer)"),
+        new Post(securityPostDenormalize: "is_granted('IS_AUTHENTICATED_FULLY') and (is_granted('ROLE_ADMIN') or ((user.isBoss() or user.isEmployee()) and user.isAssociatedWithRepairer(object.repairer)))"),
+        new Delete(security: "is_granted('ROLE_ADMIN')"),
+    ],
     normalizationContext: ['groups' => [self::DISCUSSION_READ]],
+    //    mercure: true,
     denormalizationContext: ['groups' => [self::DISCUSSION_WRITE]],
-    mercure: true,
     paginationClientEnabled: true,
-    paginationClientItemsPerPage: true,
+    paginationClientItemsPerPage: true
 )]
-#[GetCollection(security: "is_granted('IS_AUTHENTICATED_FULLY')")]
-#[Get(security: "is_granted('ROLE_ADMIN') or object.customer == user or (user.repairer and user.repairer == object.repairer) 
-or (user.repairerEmployee and user.repairerEmployee.repairer == object.repairer)")]
 #[Get(
     uriTemplate: '/discussions/{id}/set_read',
     controller: ReadMessageController::class,
-    security: 'object.customer == user or (user.repairer and user.repairer == object.repairer) or (user.repairerEmployee and user.repairerEmployee.repairer == object.repairer)',
+    security: 'object.customer == user or user.isAssociatedWithRepairer(object.repairer)',
 )]
 #[GetCollection(
-    uriTemplate: '/messages_unread',
-    controller: NumberOfMessageNotReadController::class,
+    uriTemplate: '/repairers/{repairer_id}/messages_unread',
```

Comment:
Same that this discussion:
https://github.com/RustineLibre/RustineLibre/pull/43/files#r1715080135